src/ViewerComponents/Image/ImagesContainer.svelte

```diff
@@ -0,0 +1,163 @@
+<script>
+    import MultipleImages from './MultipleImages.svelte';
+    import Loading from '../Loading.svelte';
+
+    export let requestInfos, jsonFile;
+    let promise;
+    let current_offset = 0;
+    const result_per_request = 25;
+
+    $: requestInfos && initialize();
+
+    /**
+     * Initializes the process by setting offset and calls prepareRequest.
+     */
+    function initialize() {
+        current_offset = 0;
+        prepareRequest(current_offset);
+    }
+
+    /**
+     * Prepares the request and assigns to variable 'promise'
+     *
+     * @param offset
+     */
+    function prepareRequest(offset) {
+        promise = requestData(offset);
+    }
+
+    /**
+     * Requests the data with the parameter given from parent component.
+     *
+     * @param offset
+     * @returns {Promise<unknown>}
+     */
+    async function requestData(offset) {
+        const res = await fetch(requestInfos['url'], {
+            method: requestInfos['method'],
+            body: requestInfos['gravsearch'] + `\n OFFSET ${offset}`
+        })
+
+        const json = await res.json();
+
+        // Checks if request succeeded
+        if (!res.ok) {
+            console.error(json);
+            return new Promise(() => {
+                throw new Error(`${res.status.toString()}: ${res.statusText}`);
+            })
+        }
+
+        console.log(json);
+        return json;
+    }
+
+    /**
+     * Checks if data is an empty object
+     *
+     * @param data
+     * @returns {boolean}
+     */
+    function isEmpty(data) {
+        return Object.keys(data).length === 0;
+    }
+
+    /**
+     * Increases the offset and starts a new request with the new offset.
+     */
+    function next() {
+        prepareRequest(current_offset += 1);
+    }
+
+    /**
+     * Decreases the offset and starts a new request with the new offset.
+     */
+    function previous() {
+        prepareRequest(current_offset -= 1);
+    }
+
+    /**
+     * Checks if current offset is 0.
+     *
+     * @returns {boolean}
+     */
+    function checkPrevious() {
+        return current_offset === 0;
+    }
+
+    /**
+     * Checks if data has more results by checking the flag.
+     *
+     * @param data
+     * @returns {boolean}
+     */
+    function checkNext(data) {
+        return !(data.hasOwnProperty('knora-api:mayHaveMoreResults') && data['knora-api:mayHaveMoreResults']);
```

Comment:
This seems misleading, as this returns true if there is no next, but the method name implies it would return true. I suggest to change the name or invert the logic

Reply:
Good comment, name will be changed